### ISSUE TITLE
MPIR fails to build statically with dynamic CRT

### ISSUE THREAD

**Issue by bjornpiltz** [Microsoft/vcpkg#2047]
My triplet file looks like this:
```cmake
set(VCPKG_TARGET_ARCHITECTURE x64)

# Always use the dynamic CRT.
set(VCPKG_CRT_LINKAGE dynamic)

# Prefer to link statically.
set(VCPKG_LIBRARY_LINKAGE static)

# Except for the following:
...
```

MPIR fails with the following message:
```
-- Performing post-build validation
Expected Debug,Dynamic crt linkage, but the following libs had invalid crt linkage:

    C:/vcpkg/packages/mpir_rmak/debug/lib/mpir.lib: Debug,Static

To inspect the lib files, use:
    dumpbin.exe /directives mylibfile.lib
Expected Release,Dynamic crt linkage, but the following libs had invalid crt linkage:

    C:/vcpkg/packages/mpir_rmak/lib/mpir.lib: Release,Static

To inspect the lib files, use:
    dumpbin.exe /directives mylibfile.lib
Found 2 error(s). Please correct the portfile:
    C:\vcpkg\ports\mpir\portfile.cmake
```

I can work around it by building MPIR dynamically for now.

**Comment by bjornpiltz** [comment]
The exact same error occurs with libjpeg-turbo and libmysql.

**Comment by ras0219-msft** [comment]
Thanks for reporting! I'm fixing `libjpeg-turbo` right now: d1627a0a

**Comment by ras0219-msft** [comment]
Ok, `libmysql` fixed in 2735cea57197.

Turns out MPIR was also easy to fix as well (1cdcb790)!

Both are now in master.

**Comment by bjornpiltz** [comment]
Thanks!
